### PR TITLE
Disable trackable urls in SMS integration

### PR DESCRIPTION
**Pull request by kuzmany**
[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | ✅ 
| Automated tests included? | ✅ 
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/2357 & #1873
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
PR added disable trackable urls in SMS. 
Mautic has very long trackable URL, this option allow use your own urls.

#### Steps to test this PR:
1.  Apply PR, setup Twillio integration and enable Disable trackable url option
2.  Create SMS with url
3. Send SMS through campaign action and see if your URL wasn't replace Mautic trackable url

### CODE REVIEW

**Comment by kuzmany** [comment]
Label: Ready to test

File: app/bundles/SmsBundle/Translations/en_US/messages.ini [hunk 1]
```diff
@@ -13,6 +13,8 @@ mautic.sms.config.form.sms.password="Auth Token"
 mautic.sms.config.form.sms.password.tooltip="Twilio Auth Token"
 mautic.sms.config.form.sms.sending_phone_number="Sending Phone Number"
 mautic.sms.config.form.sms.sending_phone_number.tooltip="The phone number given by your provider that you use to send and receive Text Message messages."
+mautic.sms.config.form.sms.disable_trackable_urls="Disable trackable urls"
+mautic.sms.config.form.sms.disable_trackable_urls.tooltip="This option disable trackable Urls and your SMS click stats will not work."
```
Comment (galvani) [inline app/bundles/SmsBundle/Translations/en_US/messages.ini]: Please be consistent, SMS is in capitals. URL's should too. _This option disables_

.

Comment (kuzmany) [reply app/bundles/SmsBundle/Translations/en_US/messages.ini]: Fixed

**Comment by kuzmany** [comment]
Label: Pending feedback

**Comment by Maxell92** [comment]
Waiting for tests here.

File: app/bundles/SmsBundle/EventListener/SmsSubscriber.php [hunk 2]
```diff
@@ -123,6 +132,11 @@ public function onDelete(SmsEvent $event)
      */
     public function onTokenReplacement(TokenReplacementEvent $event)
     {
+        // Disable trackable urls
+        if ($this->smsHelper->getDisableTrackableUrls()) {
+            return;
```
Comment (Maxell92) [inline app/bundles/SmsBundle/EventListener/SmsSubscriber.php]: I am not sure we can simply return here - we need to replace Lead tokes too?

**Comment by kuzmany** [comment]
Label: WIP

File: app/bundles/SmsBundle/EventListener/SmsSubscriber.php (hunk 2 above)
Comment (kuzmany) [reply app/bundles/SmsBundle/EventListener/SmsSubscriber.php]: Right, I will update it. Thanks

**Comment by kuzmany** [comment]
@Maxell92 Need hint. I don't know create test without help. 
SMS  has just TransportChainTest, should I add it  there?

**Comment by Maxell92** [comment]
Hello @kuzmany,

this new feature should be covered with tests: https://github.com/mautic/mautic/pull/5728/files#diff-84053ed64681f26013614b914966f944R148

That means at least 2 different tests for `onTokenReplacement` method - Test content when `getDisableTrackableUrls` return TRUE and when it returns FALSE.

Simple assertions should be enough :) Ping me if you need more help.

**Comment by kuzmany** [comment]
@Maxell92  Test covered . Now finally ready to test and commit

**Comment by npracht** [comment]
@Maxell92 maybe you can review your review now ? :)

**Comment by npracht** [comment]
@Maxell92 and @galvani ? :)

**Pull request merged**